src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -512,7 +512,9 @@ void minimize_local_reorders(program& p, std::map<program_node*, format::type>&
             continue;
 
         for (auto new_fmt : local_formats) {
-            if (fmt_map.at(node) != format::any && format::dimension(fmt_map.at(node)) != format::dimension(new_fmt))
+            // Avoid setting of formats which will require transform from higher rank to smaller one which requires dimension squeeze
+            // TODO: Needs to be updated once we improve layout assignment logic
+            if (fmt_map.at(node) != format::any && format::dimension(fmt_map.at(node)) > format::dimension(new_fmt))
```

Comment:
Won't this be problematic for dynamic shape? Shouldn't we add condition for (allow_new_shape_infer == false)?

Reply:
I don't think that we should disable it for new shape infer path as format selection logic is pretty much same